wolfssl/wolfcrypt/settings.h

```
@@ -4538,6 +4538,10 @@ extern void uITRON4_free(void *p) ;
 #error "If TLS is enabled please make sure either client or server is enabled."
 #endif
 
+#ifdef WOLFSSL_DEBUG
```

Do we really need to throw an error? I'm okay with that macro setting DEBUG_WOLFSSL if its not set.

---

I prefer an error over quietly auto-fixing. It is not a valid macro.

The correction will teach the user the correct macro to use.

Having an error will also ensure our own code is using the proper macro.

---

The current error in jenkins is `Cannot contact wolf-linux-cloud-node`. 

Before I retest, I'll await word if we should quietly support either `DEBUG_WOLFSSL` or `WOLFSSL_DEBUG`.

I see the merit of both, particularly since the correct one does _not_ start with `WOLFSSL_`.

---

Additionally, if we're ok to use `DEBUG_WOLFSSL` (the correct one) **as well as** `WOLFSSL_DEBUG` (the common misspelling) - then we should include some hacky logic like this to ensure full cross-use ability:

```x
#ifdef WOLFSSL_DEBUG
    #ifndef DEBUG_WOLFSSL
        #define DEBUG_WOLFSSL
    #endif
#endif

#ifdef DEBUG_WOLFSSL
    #ifndef WOLFSSL_DEBUG
        #define WOLFSSL_DEBUG
    #endif
#endif
```

Without this logic, we'll always have the issue of `WOLFSSL_DEBUG` vs `DEBUG_WOLFSSL` being used. (e.g. what if someone puts the incorrect `#define WOLFSSL_DEBUG` in `user_settings.h`?

I think it is really out of character for the codebase to support mistakes like this... but I do understand - as I too, have made the mistake on more than one occasion. 

Still, imho it sets a bad example. We probably don't want to continue and support things like allowing both of these:

`WOLFSSL_DEBUG_NONBLOCK` 
`DEBUG_WOLFSSL_NONBLOCK`

So if I'm overruled and we're going to allow `WOLFSSL_DEBUG` and `WOLFSSL_DEBUG` - I suggest that should be the limit.

---

Please don't start down this path of trying to introduce a better named macro. Its fine as-is.

---

I think we should not start error'ing on it. DEBUG_WOLFSSL is a very old macro and has been set as the public debug macro for years.